# Pauli Matrices
파울리 행렬의 아이겐밸류와 아이겐벡터를 구하시오.

$$ s_x = \begin{pmatrix} 0 & 1 \\ 1 & 0 \end{pmatrix}, \quad
s_y = \begin{pmatrix} 0 & -j \\ j & 0 \end{pmatrix}, \quad
s_z  = \begin{pmatrix} 1 & 0 \\ 0 & -1 \end{pmatrix}. $$

In [25]:
import numpy as np
import matplotlib.pyplot as plt
sx = np.array([[0,1],[1,0]])
sy = np.array([[0,1j],[1j,0]])
sz = np.array([[1,0],[0,-1]])

In [26]:
print("s_x")
print(np.linalg.eig(sx))
print("\n")
print("s_y")
print(np.linalg.eig(sy))
print("\n")
print("s_z")
print(np.linalg.eig(sz))

s_x
(array([ 1., -1.]), array([[ 0.70710678, -0.70710678],
       [ 0.70710678,  0.70710678]]))


s_y
(array([0.00000000e+00+1.j, 2.77555756e-17-1.j]), array([[ 0.70710678+0.00000000e+00j,  0.70710678+0.00000000e+00j],
       [ 0.70710678+0.00000000e+00j, -0.70710678-2.22044605e-16j]]))


s_z
(array([ 1., -1.]), array([[1., 0.],
       [0., 1.]]))


# Normal Modes
(고전역학 by Thornton & Marion 챕터 12.2 및 예제 12.3 참조) 용수철 상수가 $k$인 용수철에 연결된 두 개의 물체의 노말 모드를 구해보자. 두 개의 물체의 운동방정식은 다음으로 주어진다.$$ -m w^2 x_1 = -2kx_1 +k x_2, \\
-m w^2 x_2 = k x_1 - 2 k x_2.$$eigenfrequencies $w$는 다음의 행렬식을 통해 구할 수 있다.$$ \frac{k}{m w^2} \begin{pmatrix} 2 & -1 \\ -1 & 2 \end{pmatrix} 
\begin{pmatrix} x_1 \\ x_2 \end{pmatrix}=
\lambda \begin{pmatrix} x_1 \\ x_2 \end{pmatrix} $$

# 위의 행렬의 고유값을 구하자.

In [27]:
s = np.array([[2,-1],[-1,2]])
print(np.linalg.eig(s))

(array([3., 1.]), array([[ 0.70710678,  0.70710678],
       [-0.70710678,  0.70710678]]))


따라서 eigenfrequencies는$$ w_1 = \sqrt{\frac{k}{m}}, \quad w_2 = \sqrt{\frac{3k}{m}}. $$

위에서 구한 아이겐벡터와 아이겐밸류의 의미를 설명하여라.

아이겐벡터는 진폭을 의미한다
[3., 1.]
물체1이 물체2의 진폭보다 3배 크다는 것을 알수있다

아이겐벨류는 모드형상을 나타낸다.

[ 0.70710678,  0.70710678]은 w1일때 물체가 같은 방향으로 움직이고

[-0.70710678,  0.70710678]는 w2일때 물체가 다른 방향으로 움직인다.